friends/src/components/Form.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from "react";
+import './friends.css';
+class Form extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "",
+      age: "",
+      email: ""
+    };
+  }
+
+  onChangeHandler = event => {
+    if(event.target.type === "number"){
+      this.setState({
+        [event.target.name]: Number(event.target.value)
+      });
+    }else{
+      this.setState({
+        [event.target.name]: event.target.value
+      });
+    }
+  };
+
+  id = () => {
+    console.log(this.props);
+    this.props.friends.forEach(item => {
+      if (
+        item.name === this.state.name ||
+        item.age === this.state.age ||
+        item.email === this.state.email
```

Comment:
This seems like a lot of repetition. Maybe try a helper function? This specific one uses [some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).
```javascript
id = () => {
  const check = (item, ...keys) => {
    return keys.some(key => item[key] === this.state[key]);
  };
  this.props.friends.forEach(item => {
    if (check(item, 'name', 'age', 'email')) {
      this.props.modifyFriend(item.id, this.state);
    }
  });
};
```
You could optionally make the check function a little shorter by using an implicit return:
```javascript
const check = (item, ...keys) => (
  keys.some(key => item[key] === this.state[key])
);
```